appimageTools: usability improvements and cleanup (#191718)
NixOS/nixpkgs: doc/builders/images/appimagetools.section.md

## Wrapping {#ssec-pkgs-appimageTools-wrapping}

> To create a derivation for a appimage you can use the `wrapType2` builder, which is a wrapper around `buildFHSUserEnv` that extracts the appimage for you and generates a FHS environment to run it in. Any attribute `buildFHSUserEnv` accepts will also be valid in `wrapType2` derivations.

Reviewer: ```suggestion
To create a derivation for a appimage you can use the `wrapType2` builder. It is a wrapper around `buildFHSUserEnv` that extracts the appimage for you and generates a FHS environment to run it in. Any attribute `buildFHSUserEnv` accepts will also be valid in `wrapType2` derivations.
```

> To create a derivation for a appimage you can use the `wrapType2` builder, which is a wrapper around `buildFHSUserEnv` that extracts the appimage for you and generates a FHS environment to run it in. Any attribute `buildFHSUserEnv` accepts will also be valid in `wrapType2` derivations.

Author: > What's the reason it's called type1 and type2? Does it originate in appimage and its users will know what it means?

They're different versions of the appimage specification, see https://github.com/AppImage/AppImageSpec/blob/master/draft.md#type-1-image-format. This isn't very common knowledge at all, version one hasn't been used for quite a while now, nothing that I know of uses it.

I proposed an renaming/aliasing `wrapType2` to something along the lines of `mkAppimageDerivation` in a earlier comment on this PR to make this more intuative.

```nix
appimageTools.wrapType2 rec {
  pname = "patchwork";
  version = "4.11.4";

  src = fetchurl {
    url = "https://github.com/ssbc/patchwork/releases/download/v${version}/Patchwork-${version}-linux-x86_64.AppImage";
    sha256 = "1blsprpkvm0ws9b96gb36f0rbf8f5jgmw4x6dsb1kswr4ysf591s";
  };

  extraPkgs = pkgs: with pkgs; [ ];

  extraInstall = appimageContents: ''
    cp -r ${appimageContents}/usr/share/icons $out/share
  '';
}
```

> - `name` specifies the name of the resulting image. Note that `pname` and `version` can be specified instead as well.

Reviewer: I think it would help readability a lot to have the key statement in a brief sentence, and additional information in a subsequent paragraph. The manual is already very densely written, and it would probably help skimming it if it had more visual structure.

Adding it just here would of course make it inconsistent with the existing stuff. But maybe it would be good to start somewhere at all.

```suggestion
- `name` specifies the name of the resulting image.

   Note that `pname` and `version` can be specified instead as well.

```

Author: Good idea, I will switch over the other flags as well.

- `src` specifies the AppImage file to extract.

> - `extraInstall` specifies extra commands used for the install phase. It allows you to reference the contents of the extracted appimage using a function argument, as shown in the example.

Reviewer: ```suggestion
- `extraInstall` specifies extra commands used for the install phase.

   It allows you to reference the contents of the extracted appimage using a function argument, as shown in the example.

```

## Wrapping {#ssec-pkgs-appimageTools-wrapping}

To create a derivation for a appimage you can use the `wrapType2` builder. It is a wrapper around `buildFHSUserEnv` that extracts the appimage for you and generates a FHS environment to run it in. Any attribute `buildFHSUserEnv` accepts will also be valid in `wrapType2` derivations.

> 

Reviewer: ```suggestion

Example:

```

```nix
appimageTools.wrapType2 rec {
  pname = "patchwork";
  version = "4.11.4";

  src = fetchurl {
    url = "https://github.com/ssbc/patchwork/releases/download/v${version}/Patchwork-${version}-linux-x86_64.AppImage";
    sha256 = "1blsprpkvm0ws9b96gb36f0rbf8f5jgmw4x6dsb1kswr4ysf591s";
  };

  extraPkgs = pkgs: with pkgs; [ ];

  extraInstall = appimageContents: ''
    cp -r ${appimageContents}/usr/share/icons $out/share
  '';
}
```

- `name` specifies the name of the resulting image.

  Note that `pname` and `version` can be specified instead as well.

- `src` specifies the AppImage file to extract.

- `extraInstall` specifies extra commands used for the install phase.

  It allows you to reference the contents of the extracted appimage using a function argument, as shown in the example.

> - `extraPkgs` allows you to pass a function to include additional packages inside the FHS environment your AppImage is going to run in.

Reviewer: ```suggestion
- `extraPkgs` allows you to include additional packages inside the FHS environment your AppImage is going to run in.
   The attribute is specified as a function that takes as argument the Nixpkgs package set.
```
